saleor/static/dashboard-next/orders/components/OrderProductAddDialog/OrderProductAddDialog.tsx

```diff
@@ -71,25 +139,187 @@ const OrderProductAddDialog = withStyles(styles, {
+                                          variantIndex
+                                        ]
+                                      }
+                                      disabled={loading}
+                                      onChange={() =>
+                                        selectedVariants[productIndex][
+                                          variantIndex
+                                        ]
+                                          ? change({
+                                              target: {
+                                                name: "variants",
+                                                value: data.variants.filter(
+                                                  selectedVariant =>
+                                                    selectedVariant.id !==
+                                                    variant.id
+                                                )
+                                              }
+                                            } as any)
+                                          : change({
+                                              target: {
+                                                name: "variants",
+                                                value: [
+                                                  ...data.variants,
+                                                  variant
+                                                ]
+                                              }
+                                            } as any)
+                                      }
+                                    />
+                                  </TableCell>
+                                  <TableCell>
+                                    <div>{variant.name}</div>
+                                    <div className={classes.grayText}>
+                                      {i18n.t("SKU {{ sku }}", {
+                                        sku: variant.sku
+                                      })}
+                                    </div>
+                                  </TableCell>
+                                  <TableCell className={classes.textRight}>
+                                    <Money money={variant.price} />
+                                  </TableCell>
+                                </TableRow>
+                              )
+                            )}
+                          </React.Fragment>
+                        ),
+                        () => (
+                          <TableRow>
+                            <TableCell colSpan={4}>
+                              {i18n.t("No products matching given query")}
+                            </TableCell>
+                          </TableRow>
+                        )
+                      )}
+                    </TableBody>
+                  </Table>
+                </InfiniteScroll>
```

Comment:
You should really think about some code organizing. It's extremally hard to read IMO.
![ruciax1](https://user-images.githubusercontent.com/5078402/50686422-a29ffa00-101d-11e9-8295-61d77b98a6aa.gif)

Reply:
True, for instances there are two large `onChange` handlers for checkboxes, shouldn't these be extracted to some functions for better readability?

Reply:
You could extract functions etc to the top to make `.jsx` a lot clearer.